src/main/java/be/seeseemelk/mockbukkit/block/state/BlockStateMock.java

```diff
@@ -50,8 +52,30 @@ protected BlockStateMock(@NotNull BlockStateMock state)
 		this.block = state.isPlaced() ? state.getBlock() : null;
 	}
 
+	// region Type Checking
+	protected void checkType(Material material, Material... expected)
+	{
+		Preconditions.checkArgument(Arrays.stream(expected).anyMatch(m -> material == m), "Cannot create a " + getClass().getSimpleName() + " from " + material);
+	}
+
+	protected void checkType(Block block, Material... expected)
+	{
+		checkType(block.getType(), expected);
+	}
+
+	protected void checkType(Material material, Tag<Material> tag)
+	{
+		Preconditions.checkArgument(tag.isTagged(material), "Cannot create a " + getClass().getSimpleName() + " from " + material);
+	}
+
+	protected void checkType(Block block, Tag<Material> expected)
+	{
+		checkType(block.getType(), expected);
+	}
```

Comment:
Also, perhaps add `@NotNull` annotations here?

Reply:
Yup, they'd be good. Gonna leave out the checks for them though since they should be caught in the constructor.